src/components/popover/bl-popover.ts

```diff
@@ -0,0 +1,270 @@
+import { CSSResultGroup, html, LitElement, TemplateResult } from 'lit';
+import { customElement, property, query, state } from 'lit/decorators.js';
+import {
+  computePosition,
+  flip,
+  shift,
+  offset,
+  arrow,
+  inline,
+  autoUpdate,
+  size,
+  Middleware,
+  MiddlewareState,
+} from '@floating-ui/dom';
+import { classMap } from 'lit/directives/class-map.js';
+import style from './bl-popover.css';
+import { event, EventDispatcher } from '../../utilities/event';
+
+export type Placement =
+  | 'top-start'
+  | 'top'
+  | 'top-end'
+  | 'bottom-start'
+  | 'bottom'
+  | 'bottom-end'
+  | 'left-start'
+  | 'left'
+  | 'left-end'
+  | 'right-start'
+  | 'right'
+  | 'right-end';
+
+/**
+ * @tag bl-popover
+ * @summary Baklava Popover component
+ *
+ * @cssproperty --bl-popover-arrow-display - Sets the display of popovers arrow. Default value is `none`
+ * @cssproperty --bl-popover-background-color - Sets the background color of popover. Default value is `--bl-color-primary-background`
+ * @cssproperty --bl-popover-border-color - Sets the border color of popover. Default value is `--bl-color-primary-hover`
+ * @cssproperty --bl-popover-padding - Sets the padding of popover. Default value is `--bl-size-m`
+ * @cssproperty --bl-popover-border-radius - Sets the border radius of popover. Default value is `--bl-size-3xs`
+ * @cssproperty --bl-popover-position - Sets the position of popover. Default value is fixed
+ */
+@customElement('bl-popover')
+export default class BlPopover extends LitElement {
+  static get styles(): CSSResultGroup {
+    return [style];
+  }
+
+  @query('.popover') private popover: HTMLElement;
+  @query('.arrow') private arrow: HTMLElement;
+
+  /**
+   * Sets placement of the popover
+   */
+  @property({ type: String })
+  placement: Placement = 'bottom';
+
+  /**
+   * Target elements state
+   */
+  @state() _target: string | Element;
+
+  /**
+   * Shifts the popover along the axis
+   */
+  @property({ type: Boolean })
+  shift = false;
```

Comment:
This parameter enables the mode for keeping popover inside the viewport. But I'm hesitant about the need for having this property. In all of our components, this will be always `true` and I can not find a use-case to use this as `false`. Setting default value to `true` is also problemmatice since it's not technically possible to set a boolean attribute to `false` in HTML. That means if we set default value to `true` and if someone wants to set it to `false`, setting property via JS will be the only option.